core/src/primitive_types.rs

```diff
@@ -193,4 +193,9 @@ impl U256 {
 
 		unsafe { core::mem::transmute(data) }
 	}
-}
\ No newline at end of file
+
+
+        pub fn to_h256(&self) -> H256 {
+               unsafe { core::mem::transmute(*self) }
+        }
```

Comment:
U256 internal representation is little endian. EVM uses big endian byte order.
When converting from U256 to byte array or to H256, you need to convert little to big endian.

Reply:
And implement `From` trait for conversion